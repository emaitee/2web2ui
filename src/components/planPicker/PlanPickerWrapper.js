import React from 'react';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';
import PlanPicker from './PlanPicker';
import config from 'src/config';
import styles from './PlanPicker.module.scss';

const PlanPickerWrapper = ({ plans, pendingSubscription, subscription }) => {
  /**
   * Hide PlanPicker for users with non-self-serve plans
   */
  if (subscription && !subscription.self_serve) {
    let supportText = (
      <span>
        To make changes to your plan or billing information, <a href={`mailto:${config.contact.supportEmail}`}>contact support</a>.
      </span>
    );

    if (typeof pendingSubscription !== 'undefined') {
      supportText = (
        <span>
          You're scheduled to switch to the { pendingSubscription.name } plan on {pendingSubscription.effective_date}. If you have any questions, please <a href={`mailto:${config.contact.supportEmail}`}>contact support</a>.
        </span>
      );
    }

    return (
      <div className={styles.PlanPicker}>
        <p className={styles.ManualTitle}>Your current <strong>{ subscription.name }</strong> plan includes <strong>{ subscription.plan_volume.toLocaleString() }</strong> emails per month.</p>
        <p className={styles.ManualHelp}>{ supportText }</p>
      </div>
    );
  }

  /**
   * This component will register the a redux-form field named 'planpicker'
   * Entire plan object is stored in state
   */
  return <Field component={PlanPicker} name='planpicker' plans={plans} />;
};

PlanPickerWrapper.propTypes = {
  // List of plans in the dropdown
  plans: PropTypes.arrayOf(PropTypes.shape({
    code: PropTypes.string.isRequired,
    volume: PropTypes.number.isRequired,
    monthly: PropTypes.number.isRequired,
    overage: PropTypes.number.isRequired
  })).isRequired,

  // The account.subscription object
  subscription: PropTypes.shape({
    self_serve: PropTypes.bool,
    plan_volume: PropTypes.number,
    name: PropTypes.string
  }),

  // The account.pending_subscription object
  pendingSubscription: PropTypes.shape({
    effective_date: PropTypes.string,
    name: PropTypes.string
  })
};

export default PlanPickerWrapper;
