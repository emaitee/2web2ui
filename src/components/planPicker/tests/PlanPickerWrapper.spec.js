import { shallow } from 'enzyme';
import React from 'react';
import Downshift from 'downshift';

import PlanPickerWrapper from '../PlanPickerWrapper';

const selfServe = {
  self_serve: true,
  plan_volume: 5,
  name: 'Just 5'
};

const notSelfServe = {
  self_serve: false,
  plan_volume: 10,
  name: 'Just 10'
};

const pending = {
  effective_date: 'tomorrow',
  name: 'More than 10!'
};

it('renders PlanPicker if self-serve', () => {
  const wrapper = shallow(<PlanPickerWrapper plans={[]} subscription={selfServe}/>);
  expect(wrapper).toMatchSnapshot();
});

it('does not render PlanPicker for non-self-serve customers', () => {
  const wrapper = shallow(<PlanPickerWrapper plans={[]} subscription={notSelfServe}/>);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with a pending plan change', () => {
  const wrapper = shallow(<PlanPickerWrapper plans={[]} subscription={notSelfServe} pendingSubscription={pending}/>);
  expect(wrapper).toMatchSnapshot();
});
