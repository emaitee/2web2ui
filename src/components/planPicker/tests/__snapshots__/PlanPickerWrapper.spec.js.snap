// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render PlanPicker for non-self-serve customers 1`] = `
<div>
  <p>
    Your current 
    <strong>
      Just 10
    </strong>
     plan includes 
    <strong>
      10
    </strong>
     emails per month.
  </p>
  <p>
    <span>
      To make changes to your plan or billing information, 
      <a
        href="mailto:support@sparkpost.com"
      >
        contact support
      </a>
      .
    </span>
  </p>
</div>
`;

exports[`renders PlanPicker if self-serve 1`] = `
<Field
  component={[Function]}
  name="planpicker"
  plans={Array []}
/>
`;

exports[`renders correctly with a pending plan change 1`] = `
<div>
  <p>
    Your current 
    <strong>
      Just 10
    </strong>
     plan includes 
    <strong>
      10
    </strong>
     emails per month.
  </p>
  <p>
    <span>
      You're scheduled to switch to the 
      More than 10!
       plan on 
      tomorrow
      . If you have any questions, please 
      <a
        href="mailto:support@sparkpost.com"
      >
        contact support
      </a>
      .
    </span>
  </p>
</div>
`;
